### ISSUE TITLE
EDC COG ingestion for Trucks maps

### ISSUE THREAD

**Issue by AlessandroScremin** [eurodatacube/eodash#528]
Dear @gmilcinski @gobaRules 

I reported here the issue I got with the ingestion of Trucks maps COGs

A first try was to ingest the file as received. but got some errors.
Checked the file and noted it has no overviews and no data value was set to Nan

So I run the script to create the overviews and change the no data value to -9999

I got the error of Index 0 and other already saw in the mail.

THen anjia suggested to generate the tiles (doesn't have tiles for the first overview) and run this command
gdal_translate -co TILED=YES -co COPY_SRC_OVERVIEWS=YES --config GDAL_TIFF_OVR_BLOCKSIZE 1024 -co BLOCKXSIZE=1024 -co BLOCKYSIZE=1024 -co COMPRESS=DEFLATE E12c_HenrikFisser_MotorwayActivity_SK2_20190515.tiff <your_name_for_output_file>.tiff

So finally I deleted everyhtin from S3 and BYOC content and reingest the files, but I got this error

MotorwayActivity/E12c_HenrikFisser_(BAND)_LTTiled_20180515.tiff
> 2018-05-15 00:00:00 • FAILED
> Failed to ingested due to following errors.Unexpected error occurred while ingesting the tile.

for two files...one in attachment

Could you please check if it is somthing related to the image or something else?

thanks for the support

ALe
[E12c_HenrikFisser_MotorwayActivity_LTTiled_20180515.zip](https://github.com/eurodatacube/eodash/files/5232210/E12c_HenrikFisser_MotorwayActivity_LTTiled_20180515.zip)

**Comment by gmilcinski** [comment]
Hi @AlessandroScremin,

there are two problems with these data:
1) When uploading them to the bucket, you forgot to set "bucket owner full permission" permissions, which is why we were not able to read them (due to security settings of S3 the fact that we manage the bucket itself it does mean that we can also read them.
We have fixed this now ourselves, but make sure that you do it next time, otherwise the error will appear again.

2) The files have a "no data value" set to "NAN", where  our current spcifications expect number for "no data value" (see [docs](https://docs.sentinel-hub.com/api/latest/data/byoc/#gdal-example-command)).  
We have recognised that "nan" might be used often so we are working to improve our service and the update should be available tomorrow or on Thursday.  
Therefore, you either have to convert the tiff to have a number-based "no data value", or wait until our service supports "nan".

CC: @tomislijepcevic

**Comment by AlessandroScremin** [comment]
@gmilcinski 

thanks for the feedback...
For first point not true...every time we upload into the bucket we run the S3 setacl...same script used for Water quality...so if ti works with water quality should work also for this.

Second point is partially true, becasue I changed the no data value in the first try...but got same result as the last one i sent.

I will re run the entire processing from scratch for these files...

I will be back to you....

kr
Ale

**Comment by AlessandroScremin** [comment]
Dear @gmilcinski 
we are stucked with the access denied error as open in issue https://github.com/eurodatacube/eodash-data/issues/178
We can not try if everything is working since no upload is possible
any advice?

**Comment by gmilcinski** [comment]
I would suggest you contact Mubdi support as well as it seems this issue gies beyond our knowhow.
We are trying to identify an issue, but it will be faster if you talk with Mundi in parallel.
Some things are unfortunately out of our control, DIAS’ internal IT issues certainly so.

**Comment by gmilcinski** [comment]
Cc: @tomislijepcevic 
Do you have some idea here?

**Comment by AlessandroScremin** [comment]
@gobaRules @gmilcinski @tomislijepcevic 

New support requested:
I was finally able to ingest the cogs...but as most of them are successfully ingested some others reported the following errors:

MotorwayActivity/E12c_HenrikFisser_(BAND)_CY_20180515.tif
2018-05-15 00:00:00 • FAILED
Failed to ingested due to following errors.Band MotorwayActivity problems: TIFF has inconsistent sub-sampling factor. TIFF has inconsistent sub-sampling factor.

or

MotorwayActivity/E12c_HenrikFisser_(BAND)_IT_20180515.tif
2018-05-15 00:00:00 • FAILED
Failed to ingested due to following errors.Band MotorwayActivity problems: TIFF has unsupported bit depth for floating points.

For all of the cogs, before ingestion I run the the same following commands:
 **gdal_translate -of GTIFF ${filename}.tiff intermediate.tif -a_nodata -9999**

 **gdaladdo -r average --config GDAL_TIFF_OVR_BLOCKSIZE 1024 intermediate.tif 2 4 8 16 32**

 **gdal_translate -co TILED=YES -co COPY_SRC_OVERVIEWS=YES --config GDAL_TIFF_OVR_BLOCKSIZE 1024 -co BLOCKXSIZE=1024 -co BLOCKYSIZE=1024 -co COMPRESS=DEFLATE intermediate.tif ${filename}.tif**

What can cause the errors ??

kr

ALessandro

**Comment by gobaRules** [comment]
Hi, 

the command `gdaladdo -r average --config GDAL_TIFF_OVR_BLOCKSIZE 1024 intermediate.tif 2 4 8 16 32` should ensure that sub-sampling factor is consistent across overviews. We would need to check why this was not the case for this specific tile but I suggest we sort out https://github.com/eurodatacube/eodash/issues/555 first.

**Comment by gobaRules** [comment]
Hi Alessandro, 

we have two errors here. First the "easy" one:

1. "TIFF has unsupported bit depth for floating points" -  it seems that the data you try to ingest is not in one of the [supported formats](https://docs.sentinel-hub.com/api/latest/evalscript/v3/#sampletype). I suggest you convert the data to the FLOAT32 format with `gdal_translate` using flag `-ot `https://gdal.org/programs/gdal_translate.html#cmdoption-gdal_translate-ot

2. "TIFF has inconsistent sub-sampling factor." - this one can be a bit trickier to resolve depending on the size of your files. Can you please check the size of the files for which you have this error. If the files are smaller than 256 pixels in both directions, could you please rerun COG preparation without specifying the overview levels in `gdaladdo` command, e.g.
`gdaladdo -r average --config GDAL_TIFF_OVR_BLOCKSIZE 1024 intermediate.tif `
If the files for which you get this error are bigger than 256 pixels, we will have to check them individually.

**Comment by AlessandroScremin** [comment]
@gobaRules 
Ok I checked few data and it was as you explained. THe case of floating pooints that unique file was float64... 

I will proceed in reprocess the data and ingest these are failing...


kr
Alessandro

**Comment by AlessandroScremin** [comment]
So now it worked for the less then 256 size...

but for the float32 I got pixels bigger than the rest....se the attached image

do you have some tips?

![image](https://user-images.githubusercontent.com/65602207/95230164-ce0ed480-0801-11eb-868a-276f1c1b9deb.png)

**Comment by gobaRules** [comment]
Hi @AlessandroScremin,
I would need more information. I understand that:
1) Only one tile remains problematic, the rest of the collection works ok?  
2) For the problematic tile, the original data was in FLOAT64 and you converted it to FLOAT32, which somehow influenced the spatial resolution. 

If this holds, could you please provide:
- the original data for the problematic tile (file with float64 data)
- COG that you prepared 
- exact commands you run to prepare the COG.

**Comment by AlessandroScremin** [comment]
@gobaRules 
SOrry for the comment above...I investigated the data it self and seems that Italy 2018 has float 64 instead of 32 and also pixel size different...I redirected the solution to the Data provider.

For what concer others data now they work..I have reingested them applying suggestion provided and all tiles have been ingested. 

Only one note: I applyed the solution for the inconsistent sub-sampling factorg to the failing files and now they work. Some of them where belowe 256..other some more....applied to all of them now seems to work...

Is it correct what I do or if they are bigger then 256 could cause other unknown behaviour?

kr

ALessandro
